File: flutter/lib/infrastructure/data/dtos/ticket_statuses/ticket_status_dto.dart
import 'package:enterprise_management/infrastructure/data/dtos/projects/project_dto.dart';
import 'package:enterprise_management/infrastructure/data/dtos/tickets/ticket_dto.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'ticket_status_dto.freezed.dart';
part 'ticket_status_dto.g.dart';
@freezed
abstract class TicketStatusDto with _$TicketStatusDto {
  const factory TicketStatusDto({
    required String id,
    required String name,
    required ProjectDto projectId,
    required List<TicketDto> ticketIds,
  }) = _TicketStatusDto;

  factory TicketStatusDto.fromJson(Map<String, dynamic> json) => _$TicketStatusDtoFromJson(json);
}

File: BackEnd/Application/DTOs/TicketStatuses/TicketStatusDto.cs
using Application.DTOs.Projects;
using Application.DTOs.Tickets;

namespace Application.DTOs.TicketStatuses;

public class TicketStatusDto
{
    public Guid Id { get; set; }
    public Guid ProjectId { get; set; }
    public string Name { get; set; }
    public List<TicketDto> Tickets { get; set; }
    
}